### PR TITLE
Transport unit tests part 5 - Closes #1087

### LINKED ISSUE #1087: Complete missing / pending unit tests of the web socket transport
The following files related to the websocket based P2P transport have missing or pending tests. We should write theses tests to complete the coverage. Where it does not make sense, there should be at the very least a test skeleton.

- [x] Pending: https://github.com/LiskHQ/lisk/blob/development/modules/transport.js (https://github.com/LiskHQ/lisk/blob/development/test/unit/modules/transport.js)
  - [x] constructor
    - [x] library
  - [x] __private
    - [x] removePeer
    - [x] receiveSignatures
    - [x] receiveSignature
    - [x] receiveTransactions
    - [x] receiveTransaction
    - [x] checkInternalAccess
  - [x] Transport
    - [x] poorConsensus
    - [x] getPeers
    - [x] onBind
    - [x] onSignature
    - [x] onUnconfirmedTransaction
    - [x] onBroadcastBlock
  - [x] Transport.prototype.shared
    - [x] blocksCommon
    - [x] blocks
    - [x] list
    - [x] height
    - [x] status
    - [x] postSignature
    - [x] postSignatures
    - [x] getSignatures
    - [x] getTransactions
    - [x] postTransaction
    - [x] postTransactions
  - [x] Transport.prototype.internal
    - [x] updatePeer

*EDIT* Some of the items in the checklist were moved to a separate issue https://github.com/LiskHQ/lisk/issues/1645 because they depend on a new feature which is currently being implemented.

### PR DESCRIPTION
**Pull request by jondubois**
### What was the problem?

Missing unit test cases for transport module.

### How did I fix it?

Implemented missing test cases.

Note that some of the `postBlock` test cases where skipped intentionally because the system logic will soon be refactored to address issue https://github.com/LiskHQ/lisk/issues/1621 (block serialization).

### How to test it?

`npm run mocha test/unit/modules/transport.js`

The PR closes #1087

### CODE REVIEW

File: test/unit/modules/transport.js [hunk 1]
```diff
@@ -880,12 +934,99 @@ describe('transport', () => {
...
+					error = err;
+					result = res;
+					done();
+				});
+			});
+
+			it(
+				'should call library.schema.validate with query and definitions.WSAccessObject',
+				() => {
+					return expect(library.schema.validate.calledWith(query, definitions.WSAccessObject)).to.be.true;
```
[the remark this answers is not in the document]
Comment (karmacoma) [reply test/unit/modules/transport.js]: As discussed, we will change assertions later via separate issue / PR.

**Comment by jondubois** [comment]
@MaciejBaj I updated tests to match your changes made in https://github.com/LiskHQ/lisk/pull/1608

File: test/unit/modules/transport.js [hunk 2]
```diff
@@ -1028,550 +1220,1161 @@ describe('transport', () => {
...
+							.which.is.equal(receiveSignatureError);
 					});
 				});
+			});
 
-				describe('when query.signature is undefined', () => {
-					it('should call __private.receiveSignature');
+			describe('postSignatures', () => {
+				describe('when library.schema.validate succeeds', () => {
+					beforeEach(done => {
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: These 2 lines might be extracted and close in common `beforeEach` section of `postSignatures`
```
				query = {
							signatures: [SAMPLE_SIGNATURE_1],
						};
						__private.receiveSignatures = sinonSandbox.stub();
```

File: test/unit/modules/transport.js [hunk 3]
```diff
@@ -1028,550 +1220,1161 @@ describe('transport', () => {
...
+							constants.maxSharedTxs
+						)
+					).to.be.true;
+				});
 
-								it('should assign signatures: trs.signatures');
-							});
-						});
+				describe('when all transactions returned by modules.transactions.getMultisignatureTransactionList are multisignature transactions', () => {
+					it('should call callback with error = null and result = {success: true, signatures: signatures} where signatures contains all transactions', () => {
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: Possibly split this test and all following for 2 separate ones - result and error checking. The description will be shorter and tests will be kept more unit.

File: test/unit/modules/transport.js [hunk 4]
```diff
@@ -1028,550 +1220,1161 @@ describe('transport', () => {
...
+						error = err;
+						result = res;
+						done();
+					});
+				});
 
-			describe('postSignatures', () => {
-				describe('when query.signatures is defined', () => {
-					it('should call __private.receiveSignatures');
+				it('should call callback with error = null and result = {success: true, height: modules.system.getHeight(), broadhash: modules.system.getBroadhash(), nonce: modules.system.getNonce()}', () => {
```
Comment (MaciejBaj) [inline test/unit/modules/transport.js]: The test doesn't really correlate with the description (also it's pretty long).
So the description says: 
`should call .... with ... height: modules.system.getHeight()`, 
where the test checks the direct value
`expect(result).to.have.property('height').which.equals(headers.height);`
It should be either mocking `system.getHeight`, checking if it was called and if the value returned by it was assigned correctly or just align description to check if after headers are set up, the values are returned correctly:
```
describe('when headers are set')
   it('should call callback with result containing height = 123"')
```
Please align with other tests as well, and also separate checked values if you think it's relevant.

**Pull request merged**